File: rebecca-aiml/src/samples/network/python/consoleCallBacks/customCallBacks.py
"""
   RebeccaAIML, Artificial Intelligence Markup Language 
   C++ api and engine.
  
   Copyright (C) 2005,2006,2007 Frank Hassanabad
  
   This file is part of RebeccaAIML.
  
   RebeccaAIML is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 3 of the License, or
   (at your option) any later version.
  
   RebeccaAIML is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.
  
   You should have received a copy of the GNU General Public License
   along with this program.  If not, see <http://www.gnu.org/licenses/>.
"""

from CallBacks import *

class CustomCallBacks(CallBacks):
    __i = 0
    __numberOfCategories = 5000

    def storeGossip(self, gossip, userId, botId, endUserId):
        print "[Gossip: " + gossip + "]"

    def categoryLoaded(self, userId, botId, endUserId): 
        self.__i = self.__i + 1
        if(self.__i % self.__numberOfCategories == 0):
            print "[" + str(self.__i) + " categories loaded]"

    def filePreLoad(self, fileName, userId, botId, endUserId):
        print "[Loading      " + fileName + "]"

    def filePostLoad(self, fileName, userId, botId, endUserId):
        print "[Done loading " + fileName + "]"

    def symbolicReduction(self, symbol, userId, botId, endUserId):
        print "Symbolic reduction: " + symbol

    def infiniteSymbolicReduction(self, userId, botId, endUserId):
        print "[Infinite Symbolic reduction detected]"

    def XMLParseError(self, message, userId, botId, endUserId):
        print message

    def XMLParseWarning(self, message, userId, botId, endUserId):
        print message

    def XMLParseFatalError(self, message, userId, botId, endUserId):
        print message

    def thatStarTagSizeExceeded(self, userId, botId, endUserId):
        print "[Warning thatStar Tag Size Exceeded]"

    def topicStarTagSizeExceeded(self, userId, botId, endUserId):
        print "[Warning topicStar Tag Size Exceeded]"

    def starTagSizeExceeded(self, userId, botId, endUserId):
        print "[Warning star Tag Size Exceeded]"

    def inputTagNumericConversionError(self, message, userId, botId, endUserId):
        print "inputTagNuermicConversionError:" + message

    def inputTagSizeExceeded(self, userId, botId, endUserId):
        print "[Warning input Tag Size Exceeded]"

    def starTagNumericConversionError(self, message, userId, botId, endUserId):
        print "starTagNuermicConversionError:" + message

    def thatTagSizeExceeded(self, userId, botId, endUserId):
         print "[Warning that Tag Size Exceeded]"

    def thatTagNumericConversionError(self, message, userId, botId, endUserId):
        print "thatTagNumericConversionError:" + message

    def topicStarTagNumericConversionError(self, message, userId, botId, endUserId):
        print "topicStarTagNumericConversionError:" + message

    def thatStarTagNumericConversionError(self, message, userId, botId, endUserId):
        print "thatStarTagNumericConversionError:" + message

    def learnTagFileNotFound(self, message, userId, botId, endUserId):
        print "learnTagFileNotFound:" + message

File: rebecca-aiml/src/samples/network/java/consoleCallBacks/CustomCallBacks.java
/*
 * RebeccaAIML, Artificial Intelligence Markup Language 
 * C++ api and engine.
 *
 * Copyright (C) 2005,2006,2007 Frank Hassanabad
 *
 * This file is part of RebeccaAIML.
 *
 * RebeccaAIML is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * RebeccaAIML is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import rebecca.CallBacks;

public class CustomCallBacks extends CallBacks
{
   private int i = 0;

   @Override
    public void storeGossip(String gossip, String userId, String botId, String endUserId)
    {
        System.out.println("[Gossip: " + gossip + "]");
    }

   @Override
    public void categoryLoaded(String userId, String botId, String endUserId)
    {
        final int numberOfCategories = 5000;
       /*
        * Clever way to say if "i" is a multiple
        * of 5000 then print out the number of
        * categories loaded so far.
        */
        if(++i % numberOfCategories == 0)
        {
            System.out.println("[" + i + " categories loaded]");
        }
    }

   @Override
    public void filePreLoad(String fileName, String userId, String botId, String endUserId)
    {
        System.out.println("[Loading      " + fileName + "]");
    }

   @Override
    public void filePostLoad(String fileName, String userId, String botId, String endUserId)
    {
        System.out.println("[Done loading " + fileName + "]");
    }

   @Override
    public void symbolicReduction(String symbol, String userId, String botId, String endUserId)
    {
        System.out.println("Symbolic reduction: " + symbol);
    }

   @Override
    public void infiniteSymbolicReduction(String userId, String botId, String endUserId)
    {
        System.out.println("[Infinite Symbolic reduction detected]");
    }

   @Override
    public void XMLParseError(String message, String userId, String botId, String endUserId)
    {
        System.out.println(message);
    }

   @Override
    public void XMLParseWarning(String message, String userId, String botId, String endUserId)
    {
        System.out.println(message);
    }

   @Override
    public void XMLParseFatalError(String message, String userId, String botId, String endUserId)
    {
        System.out.println(message);
    }

   @Override
    public void thatStarTagSizeExceeded(String userId, String botId, String endUserId)
    {
        System.out.println("[Warning thatStar Tag Size Exceeded]");
    }

   @Override
    public void topicStarTagSizeExceeded(String userId, String botId, String endUserId)
    {
        System.out.println("[Warning topicStar Tag Size Exceeded]");
    }

   @Override
    public void starTagSizeExceeded(String userId, String botId, String endUserId)
    {
        System.out.println("[Warning star Tag Size Exceeded]");
    }

   @Override
    public void inputTagNumericConversionError(String message, String userId, String botId, String endUserId)
    {
        System.out.println("inputTagNuermicConversionError:" + message);
    }

   @Override
    public void inputTagSizeExceeded(String userId, String botId, String endUserId)
    {
        System.out.println("[Warning input Tag Size Exceeded]");
    }

   @Override
    public void starTagNumericConversionError(String message, String userId, String botId, String endUserId)
    {
        System.out.println("starTagNuermicConversionError:" + message);
    }

   @Override
    public void thatTagSizeExceeded(String userId, String botId, String endUserId)
    {
         System.out.println("[Warning that Tag Size Exceeded]");
    }

   @Override
    public void thatTagNumericConversionError(String message, String userId, String botId, String endUserId)
    {
        System.out.println("thatTagNumericConversionError:" + message);
    }

   @Override
    public void topicStarTagNumericConversionError(String message, String userId, String botId, String endUserId)
    {
        System.out.println("topicStarTagNumericConversionError:" + message);
    }

   @Override
    public void thatStarTagNumericConversionError(String message, String userId, String botId, String endUserId)
    {
        System.out.println("thatStarTagNumericConversionError:" + message);
    }

   @Override
    public void learnTagFileNotFound(String message, String userId, String botId, String endUserId) {
        System.out.println("learnTagFileNotFound:" + message);
    }
}
